src/main/java/net/geant/nmaas/portal/service/impl/DomainServiceImpl.java

```diff
@@ -265,12 +269,21 @@ public boolean softRemoveDomain(Long domainId) {
             domain.setName(domain.getName() + "_DELETED_" + OffsetDateTime.now());
             domain.setCodename(domain.getCodename() + "_DELETED_" + OffsetDateTime.now());
 			removeAllUsersFromDomain(domain);
-            triggerApplicationsUninstall();
+			removeDomainFromAllGroups(domain);
+			triggerApplicationsUninstall();
 			domainRepository.save(domain);
 			return true;
 		}).orElse(false);
 	}
 
+	@Override
+	public void removeDomainFromAllGroups(Domain domain) {
+		for (int i = 0; i < domain.getGroups().size(); i++) {
```

Comment:
I think that there should be a better way of implementing this removal of object from a list using Iterator. Could you check this out?